### ISSUE TITLE
SonarCloud does not gather coverage anymore

### ISSUE THREAD

**Issue by echebbi** [pitest/pitclipse#105]
## Bug description

SonarCloud always report 0% coverage

## Expected behavior

SonarCloud should report the right coverage.

## How to reproduce

Build an internal branch/PR on Travis CI.

## Additional context

It looks like Sonar team has deprecated usage of JaCoCo's `.exec` files and doesn't take them into account anymore. [Here](https://community.sonarsource.com/t/coverage-test-data-importing-jacoco-coverage-report-in-xml-format/12151) they advise to rely on XML reports instead, hopefully we can easily generate them.

**Comment by LorenzoBettini** [comment]
@echebbi the jacoco-maven-plugin provides the goal `report` to do that (automatically bound to the `verify` phase): you just have to enable such a goal in the configuration of the plugin.

**Comment by echebbi** [comment]
Yes, that's what is advised by the Sonar team. I gave it a try but can't make it work: either the XML report is not generated or I face errors like `Unknown block type c0` or `malformed input around byte 2`. 

Upgrading the JaCoCo plug-in to the latest version didn't help so I must be doing something wrong. I'll take a deeper look when I'll have more time.

**Comment by LorenzoBettini** [comment]
I can re-enable SonarCloud analysis from GitHub Actions, but that requires using Java 11. I hope that doesn't break anything ;)

**Comment by LorenzoBettini** [comment]
By the way, to do this, @echebbi I'd need you to create a secret in this repository called named `SONAR_TOKEN` with the value of your SonarCloud token (since the analysis is hosted on your SonarCloud's account).

**Comment by echebbi** [comment]
@LorenzoBettini `SONAR_TOKEN` is now available on this repository.

By the way, what will be the consequences of migrating to Java 11 for the end users? I guess that they would at least need Java 11 to run the plug-in?

**Comment by LorenzoBettini** [comment]
> @LorenzoBettini `SONAR_TOKEN` is now available on this repository.
> 

Thanks! I'm about to try that

> By the way, what will be the consequences of migrating to Java 11 for the end-users? I guess that they would at least need Java 11 to run the plug-in?

No consequence: the important thing is the required Java version in our bundles, which is Java 8. End-users need at least Java 8, they DON'T need Java 11. We only need to use Java 11 for building (actually only for running the SonarCloud analysis). The bytecode is generated to require Java 8. By the way, using Java 11 for building also has the benefits of switching to Tycho 2 and of building and testing against Eclipse from 2020-09 on.
